### PR TITLE
Stardog .NET example 

### PR DESCRIPTION
**Pull request by joephayes**
Stardog .NET example using the [TrinityRDF](https://trinity-rdf.net/) library.

- Updated README with data requirements and instructions on how to build and run .NET sample
- Added gradle build targets - `compileDotnet`and `runDotnet`
- Fixed compile error in service example
- Updates gradle wrapper to 4.10 (from 2.2)

This requires .NET Core to build and run. While you can use the latest version of [.NET Core](https://dotnet.microsoft.com/download) to build the sample you also need to install the [.NET Core v. 2.1 runtime](https://dotnet.microsoft.com/download/dotnet-core/2.1)

**N.B.** I give instructions about the data that should be loaded into Stardog for the sample to work correctly. Let me know if this should be integrated into the embedded server used by some of the other samples.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -6,10 +6,11 @@ task wrapper(type: Wrapper) {
 
 ext {
 	stardogVersion = project.hasProperty("stardogVersion") ? project.getProperty("stardogVersion") : '7.0.2'
-	projectVersion = '0.0.5'
+        projectVersion = '0.0.6'
 
 	distPath = "${projectDir}/dist"
 	stardogRepo = project.hasProperty("stardogRepo") ? project.getProperty("stardogRepo") : "http://maven.stardog.com"
+        stardogLib = project.hasProperty("stardogLib") ? project.getProperty("stardogLib") : "."
 }
```
Comment (snowell) [inline build.gradle:15]: Gotta say that the indentation here bugs me a bit. Unless it's just Github's editor

File: readme.md [hunk 2]
```diff
@@ -27,6 +27,28 @@ program, but you can specify the fully-qualified class name of any of the other
...
 ```
 
+### .NET Examples
+
+To compile the .NET examples, you will need to install [.NET Core](https://dotnet.microsoft.com/download). Make sure that the `dotnet` tool is on your `PATH`
+
+Once you have installed .NET Core, you can build the sample with the following command:
+
+```bash
+gradle compileDotnet
```
Comment (snowell) [inline readme.md]: Should we use gradle wrapper like we generally do, just to make sure differing gradle versions don't come back to bite us?

File: examples/dotnet/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Stardog .NET Examples
+
+The example project demonstrates how to connect to a Stardog server using [TrinityRDF](https://trinity-rdf.net/) and execute queries using SPARQL and LINQ.
+
+TrinityRDF provides an object mapper layer similar to Entity Framework for RDF data. The ontologies the sample uses can be found in the solution's `Ontologies/` folder. Trinity RDF discovers the ontologies in the solution's `ontologies.config` file. TrinityRDF uses the ontologies to generate mapping code - `Ontologies/Ontologies.g.cs`. A developer uses this generated code to create object models (see the files under the solution's `ObjectModels/` folder) with annotations that connect the C# object to the RDF data. These object models then allow .NET developers to create powerful applications without the need to learn SPARQL!
+
+## Prerequisites
+* Stardog with a database named "music" populated the data from [music_schema.ttl](./examples/dotnet/TrinityConsoleSample/Ontologies/music_schema.ttl) and [music.ttl.gz](https://github.com/stardog-union/stardog-tutorials/blob/master/music/music.ttl.gz) in a named graph called `http://stardog.com/tutorial`
```
Comment (snowell) [inline examples/dotnet/README.md]: AFAICT this link is broken and possibly would be even after a merge

Comment (snowell) [reply examples/dotnet/README.md]: To clarify: the music_schema.ttl link

**Review by snowell — CHANGES_REQUESTED** [review]
Using a local checkout on my mac, `gradle compileDotnet` and `gradle runDotnet` are both failing:

```
✘ stephen:dotnet [ jph-dotnet-example | …3 ] ➭ gradle compileDotnet

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/stephen/stardog-examples/examples/dotnet/build.gradle' line: 2

* What went wrong:
A problem occurred evaluating project ':examples:dotnet'.
> Could not get unknown property 'solution' for project ':examples:dotnet' of type org.gradle.api.Project.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
```

File: readme.md (hunk 2 above)
Comment (joephayes) [reply readme.md]: my bad - tabs vs spaces. fixed.

File: build.gradle (hunk 1 above)
Comment (joephayes) [reply build.gradle:15]: my bad - tabs vs. spaces - fixed

**Comment by joephayes** [comment]
@snowell RE: the build failure. please pull the latest - I had forgotten a properties file (hidden by gitignore) which defined the `solution` property.

File: readme.md (hunk 2 above)
Comment (joephayes) [reply readme.md]: Switched - updated gradle wrapper and updated the README files

File: examples/dotnet/README.md (hunk 3 above)
Comment (joephayes) [reply examples/dotnet/README.md]: removed the link

**Review by snowell — APPROVED** [review]

**Pull request merged**